railties/lib/rails/commands/server/server_command.rb

```diff
@@ -243,11 +243,13 @@ def log_to_stdout?
         end
 
         def pid
-          File.expand_path(options[:pid] || ENV.fetch("PIDFILE", DEFAULT_PIDFILE))
+          default_pidfile = environment == "development" ? DEFAULT_PIDFILE : nil
+          pid = options[:pid] || ENV["PIDFILE"] || default_pidfile
```

Comment:
If someone specify they `PIDFILE` env or the pid option we should generate it regardless of the environment.

Reply:
> If someone specify they PIDFILE env or the pid option we should generate it regardless of the environment.

@rafaelfranca I'm sorry if I got it somehow wrong, but isn't that what happens? :sweat_smile: 

The assignment to `pid` is evaluated regardless of the environment. So if the user sets either the option or `ENV["PIDFILE"]`, their respective values will be assigned to `pid`
```
pid = options[:pid] || ENV["PIDFILE"] || default_pidfile # < always non-nil if e.g. PIDFILE is assigned
```
Only the 'fallback' `default_pidfile` is unassigned in production, everything else stays the same as before. Open to suggestions on how to make the code clearer :slightly_smiling_face:

Reply:
lol, right. Sorry. I got it wrong